process/coordinator/process_test.go

```diff
@@ -859,17 +867,88 @@ func TestTransactionCoordinator_CreateMbsAndProcessTransactionsFromMe(t *testing
 		return true
 	}
 
+	marshalizer := &mock.MarshalizerMock{}
+	hasher := &mock.HasherMock{}
+	for i := uint32(0); i < nrShards; i++ {
+		strCache := process.ShardCacherIdentifier(0, i)
+		newTx := &transaction.Transaction{GasLimit: uint64(i)}
```

Comment:
Why is `GasLimit` equal to the shard ID?

Reply:
create transaction with different hash

Reply:
Ok, no problem. Curious though, why not use the transaction nonce?